[C,test/auth.js,1244]                             client.disconnect();
[C,test/auth.js,1245]                             server.stop(done);
[C,test/auth.js,1246]                         };
[C,test/auth.js,1247] 
[C,test/auth.js,1248]                         client.subscribe('/', handler, (err) => {
[C,test/auth.js,1249] 
@@ -1230,30 +1259,31 @@ describe('authentication', () => {
[C,test/auth.js,1259] 
[C,test/auth.js,1260]             const server = new Hapi.Server();
[C,test/auth.js,1261]             server.connection();
[C,test/auth.js,1262] 
[C,test/auth.js,1263]             server.auth.scheme('custom', internals.implementation);
[C,test/auth.js,1264]             server.auth.strategy('default', 'custom', true);
[C,test/auth.js,1265] 
[C,test/auth.js,1266]             server.register({ register: Nes, options: { auth: { type: 'direct', password: 'password' } } }, (err) => {
[C,test/auth.js,1267] 
[C,test/auth.js,1268]                 expect(err).to.not.exist();
[C,test/auth.js,1269] 
[C,test/auth.js,1270]                 server.subscription('/', { auth: { scope: ['b', 'a'] } });
[C,test/auth.js,1271] 
[C,test/auth.js,1272]                 server.start((err) => {
[C,test/auth.js,1273] 
[A,test/auth.js,1274]                     expect(err).to.not.exist();
[C,test/auth.js,1275]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/auth.js,1276]                     client.connect({ auth: { headers: { authorization: 'Custom ed' } } }, (err) => {
[C,test/auth.js,1277] 
[C,test/auth.js,1278]                         expect(err).to.not.exist();
[C,test/auth.js,1279]                         const handler = (update) => {
[C,test/auth.js,1280] 
[C,test/auth.js,1281]                             expect(update).to.equal('heya');
[C,test/auth.js,1282]                             expect(client.subscriptions()).to.deep.equal(['/']);
[C,test/auth.js,1283] 
[C,test/auth.js,1284]                             client.disconnect();
[C,test/auth.js,1285]                             server.stop(done);
[C,test/auth.js,1286]                         };
[C,test/auth.js,1287] 
[C,test/auth.js,1288]                         client.subscribe('/', handler, (err) => {
[C,test/auth.js,1289] 
@@ -1269,30 +1299,31 @@ describe('authentication', () => {
[C,test/auth.js,1299] 
[C,test/auth.js,1300]             const server = new Hapi.Server();
[C,test/auth.js,1301]             server.connection();
[C,test/auth.js,1302] 
[C,test/auth.js,1303]             server.auth.scheme('custom', internals.implementation);
[C,test/auth.js,1304]             server.auth.strategy('default', 'custom', true);
[C,test/auth.js,1305] 
[C,test/auth.js,1306]             server.register({ register: Nes, options: { auth: { type: 'direct', password: 'password' } } }, (err) => {
[C,test/auth.js,1307] 
[C,test/auth.js,1308]                 expect(err).to.not.exist();
[C,test/auth.js,1309] 
[C,test/auth.js,1310]                 server.subscription('/{id}', { auth: { scope: ['b', '{id}'] } });
[C,test/auth.js,1311] 
[C,test/auth.js,1312]                 server.start((err) => {
[C,test/auth.js,1313] 
[A,test/auth.js,1314]                     expect(err).to.not.exist();
[C,test/auth.js,1315]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/auth.js,1316]                     client.connect({ auth: { headers: { authorization: 'Custom ed' } } }, (err) => {
[C,test/auth.js,1317] 
[C,test/auth.js,1318]                         expect(err).to.not.exist();
[C,test/auth.js,1319]                         const handler = (update) => {
[C,test/auth.js,1320] 
[C,test/auth.js,1321]                             expect(update).to.equal('heya');
[C,test/auth.js,1322]                             expect(client.subscriptions()).to.deep.equal(['/5']);
[C,test/auth.js,1323] 
[C,test/auth.js,1324]                             client.disconnect();
[C,test/auth.js,1325]                             server.stop(done);
[C,test/auth.js,1326]                         };
[C,test/auth.js,1327] 
[C,test/auth.js,1328]                         client.subscribe('/5', handler, (err) => {
[C,test/auth.js,1329] 
@@ -1308,30 +1339,31 @@ describe('authentication', () => {
[C,test/auth.js,1339] 
[C,test/auth.js,1340]             const server = new Hapi.Server();
[C,test/auth.js,1341]             server.connection();
[C,test/auth.js,1342] 
[C,test/auth.js,1343]             server.auth.scheme('custom', internals.implementation);
[C,test/auth.js,1344]             server.auth.strategy('default', 'custom', true);
[C,test/auth.js,1345] 
[C,test/auth.js,1346]             server.register({ register: Nes, options: { auth: { type: 'direct', password: 'password' } } }, (err) => {
[C,test/auth.js,1347] 
[C,test/auth.js,1348]                 expect(err).to.not.exist();
[C,test/auth.js,1349] 
[C,test/auth.js,1350]                 server.subscription('/', { auth: { scope: 'b' } });
[C,test/auth.js,1351] 
[C,test/auth.js,1352]                 server.start((err) => {
[C,test/auth.js,1353] 
[A,test/auth.js,1354]                     expect(err).to.not.exist();
[C,test/auth.js,1355]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/auth.js,1356]                     client.connect({ auth: { headers: { authorization: 'Custom john' } } }, (err) => {
[C,test/auth.js,1357] 
[C,test/auth.js,1358]                         expect(err).to.not.exist();
[C,test/auth.js,1359]                         client.subscribe('/', Hoek.ignore, (err) => {
[C,test/auth.js,1360] 
[C,test/auth.js,1361]                             expect(err).to.exist();
[C,test/auth.js,1362]                             expect(err.message).to.equal('Insufficient scope to subscribe, expected any of: b');
[C,test/auth.js,1363]                             expect(client.subscriptions()).to.deep.equal([]);
[C,test/auth.js,1364] 
[C,test/auth.js,1365]                             client.disconnect();
[C,test/auth.js,1366]                             server.stop(done);
[C,test/auth.js,1367]                         });
[C,test/auth.js,1368]                     });
[C,test/auth.js,1369]                 });
@@ -1342,30 +1374,31 @@ describe('authentication', () => {
[C,test/auth.js,1374] 
[C,test/auth.js,1375]             const server = new Hapi.Server();
[C,test/auth.js,1376]             server.connection();
[C,test/auth.js,1377] 
[C,test/auth.js,1378]             server.auth.scheme('custom', internals.implementation);
[C,test/auth.js,1379]             server.auth.strategy('default', 'custom', true);
[C,test/auth.js,1380] 
[C,test/auth.js,1381]             server.register({ register: Nes, options: { auth: { type: 'direct', password: 'password' } } }, (err) => {
[C,test/auth.js,1382] 
[C,test/auth.js,1383]                 expect(err).to.not.exist();
[C,test/auth.js,1384] 
[C,test/auth.js,1385]                 server.subscription('/', { auth: { scope: 'x' } });
[C,test/auth.js,1386] 
[C,test/auth.js,1387]                 server.start((err) => {
[C,test/auth.js,1388] 
[A,test/auth.js,1389]                     expect(err).to.not.exist();
[C,test/auth.js,1390]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/auth.js,1391]                     client.connect({ auth: { headers: { authorization: 'Custom ed' } } }, (err) => {
[C,test/auth.js,1392] 
[C,test/auth.js,1393]                         expect(err).to.not.exist();
[C,test/auth.js,1394]                         client.subscribe('/', Hoek.ignore, (err) => {
[C,test/auth.js,1395] 
[C,test/auth.js,1396]                             expect(err).to.exist();
[C,test/auth.js,1397]                             expect(err.message).to.equal('Insufficient scope to subscribe, expected any of: x');
[C,test/auth.js,1398]                             expect(client.subscriptions()).to.deep.equal([]);
[C,test/auth.js,1399] 
[C,test/auth.js,1400]                             client.disconnect();
[C,test/auth.js,1401]                             server.stop(done);
[C,test/auth.js,1402]                         });
[C,test/auth.js,1403]                     });
[C,test/auth.js,1404]                 });
@@ -1376,30 +1409,31 @@ describe('authentication', () => {
[C,test/auth.js,1409] 
[C,test/auth.js,1410]             const server = new Hapi.Server();
[C,test/auth.js,1411]             server.connection();
[C,test/auth.js,1412] 
[C,test/auth.js,1413]             server.auth.scheme('custom', internals.implementation);
[C,test/auth.js,1414]             server.auth.strategy('default', 'custom', true);
[C,test/auth.js,1415] 
[C,test/auth.js,1416]             server.register({ register: Nes, options: { auth: { type: 'direct', password: 'password' } } }, (err) => {
[C,test/auth.js,1417] 
[C,test/auth.js,1418]                 expect(err).to.not.exist();
[C,test/auth.js,1419] 
[C,test/auth.js,1420]                 server.subscription('/', { auth: { scope: 'x' } });
[C,test/auth.js,1421] 
[C,test/auth.js,1422]                 server.start((err) => {
[C,test/auth.js,1423] 
[A,test/auth.js,1424]                     expect(err).to.not.exist();
[C,test/auth.js,1425]                     const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/auth.js,1426]                     client.connect({ auth: { headers: { authorization: 'Custom app' } } }, (err) => {
[C,test/auth.js,1427] 
[C,test/auth.js,1428]                         expect(err).to.not.exist();
[C,test/auth.js,1429]                         client.subscribe('/', Hoek.ignore, (err) => {
[C,test/auth.js,1430] 
[C,test/auth.js,1431]                             expect(err).to.exist();
[C,test/auth.js,1432]                             expect(err.message).to.equal('Insufficient scope to subscribe, expected any of: x');
[C,test/auth.js,1433]                             expect(client.subscriptions()).to.deep.equal([]);
[C,test/auth.js,1434] 
[C,test/auth.js,1435]                             client.disconnect();
[C,test/auth.js,1436]                             server.stop(done);
[C,test/auth.js,1437]                         });
[C,test/auth.js,1438]                     });
[C,test/auth.js,1439]                 });
@@ -31,31 +31,34 @@ describe('Browser', () => {
[C,test/client.js,31]             it('fails to connect', (done) => {
[C,test/client.js,32] 
[C,test/client.js,33]                 const client = new Nes.Client('http://nosuchexamplecom');
[C,test/client.js,34] 
[C,test/client.js,35]                 client.connect((err) => {
[C,test/client.js,36] 
[C,test/client.js,37]                     expect(err).to.exist();
[C,test/client.js,38]                     expect(err.message).to.match(/getaddrinfo ENOTFOUND/);
[C,test/client.js,39]                     done();
[C,test/client.js,40]                 });
[C,test/client.js,41]             });
[C,test/client.js,42] 
[C,test/client.js,43]             it('handles error before open events', (done) => {
[C,test/client.js,44] 
[C,test/client.js,45]                 const client = new Nes.Client('http://nosuchexamplecom');
[D,test/client.js,46]                 client.onError = function (err) { };
[A,test/client.js,46]                 client.onError = function (err) {
[A,test/client.js,47] 
[A,test/client.js,48]                     expect(err).to.exist();
[A,test/client.js,49]                 };
[C,test/client.js,50] 
[C,test/client.js,51]                 client.connect((err) => {
[C,test/client.js,52] 
[C,test/client.js,53]                     expect(err).to.exist();
[C,test/client.js,54]                     expect(err.message).to.equal('test');
[C,test/client.js,55]                     done();
[C,test/client.js,56]                 });
[C,test/client.js,57] 
[C,test/client.js,58]                 client._ws.emit('error', new Error('test'));
[C,test/client.js,59]                 client._ws.emit('open');
[C,test/client.js,60]             });
[C,test/client.js,61]         });
[C,test/client.js,62] 
[C,test/client.js,63]         describe('disconnect()', () => {
[C,test/client.js,64] 
@@ -65,92 +68,96 @@ describe('Browser', () => {
[C,test/client.js,68] 
[C,test/client.js,69]                 client.disconnect();
[C,test/client.js,70]                 done();
[C,test/client.js,71]             });
[C,test/client.js,72] 
[C,test/client.js,73]             it('ignores when client is disconnecting', (done) => {
[C,test/client.js,74] 
[C,test/client.js,75]                 const server = new Hapi.Server();
[C,test/client.js,76]                 server.connection();
[C,test/client.js,77]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,78] 
[C,test/client.js,79]                     expect(err).to.not.exist();
[C,test/client.js,80] 
[C,test/client.js,81]                     server.start((err) => {
[C,test/client.js,82] 
[A,test/client.js,83]                         expect(err).to.not.exist();
[C,test/client.js,84]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,85]                         client.connect(() => {
[C,test/client.js,86] 
[C,test/client.js,87]                             client.disconnect();
[C,test/client.js,88]                             client.disconnect();
[C,test/client.js,89]                             server.stop(done);
[C,test/client.js,90]                         });
[C,test/client.js,91]                     });
[C,test/client.js,92]                 });
[C,test/client.js,93]             });
[C,test/client.js,94] 
[C,test/client.js,95]             it('allows closing from inside request callback', (done) => {
[C,test/client.js,96] 
[C,test/client.js,97]                 const server = new Hapi.Server();
[C,test/client.js,98]                 server.connection();
[C,test/client.js,99]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,100] 
[C,test/client.js,101]                     expect(err).to.not.exist();
[C,test/client.js,102] 
[C,test/client.js,103]                     server.route({
[C,test/client.js,104]                         method: 'GET',
[C,test/client.js,105]                         path: '/',
[C,test/client.js,106]                         handler: function (request, reply) {
[C,test/client.js,107] 
[C,test/client.js,108]                             return reply('hello');
[C,test/client.js,109]                         }
[C,test/client.js,110]                     });
[C,test/client.js,111] 
[C,test/client.js,112]                     server.start((err) => {
[C,test/client.js,113] 
[A,test/client.js,114]                         expect(err).to.not.exist();
[C,test/client.js,115]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,116]                         client.connect(() => {
[C,test/client.js,117] 
[C,test/client.js,118]                             client.request('/', (err, payload, statusCode, headers) => {
[C,test/client.js,119] 
[C,test/client.js,120]                                 expect(err).to.not.exist();
[C,test/client.js,121]                                 client.disconnect();
[C,test/client.js,122]                                 setTimeout(() => server.stop(done), 100);
[C,test/client.js,123]                             });
[C,test/client.js,124]                         });
[C,test/client.js,125]                     });
[C,test/client.js,126]                 });
[C,test/client.js,127]             });
[C,test/client.js,128]         });
[C,test/client.js,129] 
[C,test/client.js,130]         describe('_reconnect()', () => {
[C,test/client.js,131] 
[C,test/client.js,132]             it('reconnects automatically', (done) => {
[C,test/client.js,133] 
[C,test/client.js,134]                 const server = new Hapi.Server();
[C,test/client.js,135]                 server.connection();
[C,test/client.js,136]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,137] 
[C,test/client.js,138]                     expect(err).to.not.exist();
[C,test/client.js,139] 
[C,test/client.js,140]                     server.start((err) => {
[C,test/client.js,141] 
[A,test/client.js,142]                         expect(err).to.not.exist();
[C,test/client.js,143]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,144] 
[C,test/client.js,145]                         let e = 0;
[C,test/client.js,146]                         client.onError = function (err) {
[C,test/client.js,147] 
[A,test/client.js,148]                             expect(err).to.exist();
[C,test/client.js,149]                             ++e;
[C,test/client.js,150]                         };
[C,test/client.js,151] 
[C,test/client.js,152]                         let c = 0;
[C,test/client.js,153]                         client.onConnect = function () {
[C,test/client.js,154] 
[C,test/client.js,155]                             ++c;
[C,test/client.js,156]                             if (c === 2) {
[C,test/client.js,157]                                 expect(e).to.equal(0);
[C,test/client.js,158]                                 client.disconnect();
[C,test/client.js,159]                                 server.stop(done);
[C,test/client.js,160]                             }
[C,test/client.js,161]                         };
[C,test/client.js,162] 
[C,test/client.js,163]                         expect(c).to.equal(0);
@@ -166,35 +173,37 @@ describe('Browser', () => {
[C,test/client.js,173]                         });
[C,test/client.js,174]                     });
[C,test/client.js,175]                 });
[C,test/client.js,176]             });
[C,test/client.js,177] 
[C,test/client.js,178]             it('does not reconnect automatically', (done) => {
[C,test/client.js,179] 
[C,test/client.js,180]                 const server = new Hapi.Server();
[C,test/client.js,181]                 server.connection();
[C,test/client.js,182]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,183] 
[C,test/client.js,184]                     expect(err).to.not.exist();
[C,test/client.js,185] 
[C,test/client.js,186]                     server.start((err) => {
[C,test/client.js,187] 
[A,test/client.js,188]                         expect(err).to.not.exist();
[C,test/client.js,189]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,190] 
[C,test/client.js,191]                         let e = 0;
[C,test/client.js,192]                         client.onError = function (err) {
[C,test/client.js,193] 
[A,test/client.js,194]                             expect(err).to.exist();
[C,test/client.js,195]                             ++e;
[C,test/client.js,196]                         };
[C,test/client.js,197] 
[C,test/client.js,198]                         let c = 0;
[C,test/client.js,199]                         client.onConnect = function () {
[C,test/client.js,200] 
[C,test/client.js,201]                             ++c;
[C,test/client.js,202]                         };
[C,test/client.js,203] 
[C,test/client.js,204]                         expect(c).to.equal(0);
[C,test/client.js,205]                         expect(e).to.equal(0);
[C,test/client.js,206]                         client.connect({ reconnect: false, delay: 10 }, () => {
[C,test/client.js,207] 
[C,test/client.js,208]                             expect(c).to.equal(1);
[C,test/client.js,209]                             expect(e).to.equal(0);
@@ -208,30 +217,31 @@ describe('Browser', () => {
[C,test/client.js,217]                         });
[C,test/client.js,218]                     });
[C,test/client.js,219]                 });
[C,test/client.js,220]             });
[C,test/client.js,221] 
[C,test/client.js,222]             it('overrides max delay', (done) => {
[C,test/client.js,223] 
[C,test/client.js,224]                 const server = new Hapi.Server();
[C,test/client.js,225]                 server.connection();
[C,test/client.js,226]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,227] 
[C,test/client.js,228]                     expect(err).to.not.exist();
[C,test/client.js,229] 
[C,test/client.js,230]                     server.start((err) => {
[C,test/client.js,231] 
[A,test/client.js,232]                         expect(err).to.not.exist();
[C,test/client.js,233]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,234] 
[C,test/client.js,235]                         let c = 0;
[C,test/client.js,236]                         const now = Date.now();
[C,test/client.js,237]                         client.onConnect = function () {
[C,test/client.js,238] 
[C,test/client.js,239]                             ++c;
[C,test/client.js,240] 
[C,test/client.js,241]                             if (c < 6) {
[C,test/client.js,242]                                 client._ws.close();
[C,test/client.js,243]                                 return;
[C,test/client.js,244]                             }
[C,test/client.js,245] 
[C,test/client.js,246]                             expect(Date.now() - now).to.be.below(150);
[C,test/client.js,247] 
@@ -242,36 +252,38 @@ describe('Browser', () => {
[C,test/client.js,252]                         client.connect({ delay: 10, maxDelay: 11 }, () => { });
[C,test/client.js,253]                     });
[C,test/client.js,254]                 });
[C,test/client.js,255]             });
[C,test/client.js,256] 
[C,test/client.js,257]             it('reconnects automatically (with errors)', (done) => {
[C,test/client.js,258] 
[C,test/client.js,259]                 const server = new Hapi.Server();
[C,test/client.js,260]                 server.connection();
[C,test/client.js,261]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,262] 
[C,test/client.js,263]                     expect(err).to.not.exist();
[C,test/client.js,264] 
[C,test/client.js,265]                     server.start((err) => {
[C,test/client.js,266] 
[A,test/client.js,267]                         expect(err).to.not.exist();
[C,test/client.js,268]                         const url = 'http://localhost:' + server.info.port;
[C,test/client.js,269]                         const client = new Nes.Client(url);
[C,test/client.js,270] 
[C,test/client.js,271]                         let e = 0;
[C,test/client.js,272]                         client.onError = function (err) {
[C,test/client.js,273] 
[A,test/client.js,274]                             expect(err).to.exist();
[C,test/client.js,275]                             ++e;
[C,test/client.js,276]                             client._url = 'http://localhost:' + server.info.port;
[C,test/client.js,277]                         };
[C,test/client.js,278] 
[C,test/client.js,279]                         let c = 0;
[C,test/client.js,280]                         client.onConnect = function () {
[C,test/client.js,281] 
[C,test/client.js,282]                             ++c;
[C,test/client.js,283] 
[C,test/client.js,284]                             if (c < 5) {
[C,test/client.js,285]                                 client._ws.close();
[C,test/client.js,286] 
[C,test/client.js,287]                                 if (c === 3) {
[C,test/client.js,288]                                     client._url = 'http://invalid';
[C,test/client.js,289]                                 }
@@ -301,57 +313,59 @@ describe('Browser', () => {
[C,test/client.js,313] 
[C,test/client.js,314]                     server.route({
[C,test/client.js,315]                         method: 'GET',
[C,test/client.js,316]                         path: '/',
[C,test/client.js,317]                         handler: function (request, reply) {
[C,test/client.js,318] 
[C,test/client.js,319]                             setTimeout(() => {
[C,test/client.js,320] 
[C,test/client.js,321]                                 return reply('hello');
[C,test/client.js,322]                             }, 10);
[C,test/client.js,323]                         }
[C,test/client.js,324]                     });
[C,test/client.js,325] 
[C,test/client.js,326]                     server.start((err) => {
[C,test/client.js,327] 
[A,test/client.js,328]                         expect(err).to.not.exist();
[C,test/client.js,329]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,330]                         client.connect(() => {
[C,test/client.js,331] 
[C,test/client.js,332]                             client.request('/', (err, payload, statusCode, headers) => {
[C,test/client.js,333] 
[C,test/client.js,334]                                 expect(err).to.exist();
[C,test/client.js,335]                                 expect(err.message).to.equal('Request failed - server disconnected');
[C,test/client.js,336] 
[C,test/client.js,337]                                 server.stop(done);
[C,test/client.js,338]                             });
[C,test/client.js,339] 
[C,test/client.js,340]                             client.disconnect();
[C,test/client.js,341]                         });
[C,test/client.js,342]                     });
[C,test/client.js,343]                 });
[C,test/client.js,344]             });
[C,test/client.js,345] 
[C,test/client.js,346]             it('times out', (done) => {
[C,test/client.js,347] 
[C,test/client.js,348]                 const server = new Hapi.Server();
[C,test/client.js,349]                 server.connection();
[C,test/client.js,350]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,351] 
[C,test/client.js,352]                     expect(err).to.not.exist();
[C,test/client.js,353] 
[C,test/client.js,354]                     server.start((err) => {
[C,test/client.js,355] 
[A,test/client.js,356]                         expect(err).to.not.exist();
[C,test/client.js,357]                         server.connections[0].plugins.nes._listener._wss.handleUpgrade = function () { };
[C,test/client.js,358] 
[C,test/client.js,359]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,360] 
[C,test/client.js,361]                         let c = 0;
[C,test/client.js,362]                         client.onConnect = function () {
[C,test/client.js,363] 
[C,test/client.js,364]                             ++c;
[C,test/client.js,365]                         };
[C,test/client.js,366] 
[C,test/client.js,367]                         let e = 0;
[C,test/client.js,368]                         client.onError = function (err) {
[C,test/client.js,369] 
[C,test/client.js,370]                             ++e;
[C,test/client.js,371]                             expect(err).to.exist();
@@ -373,30 +387,31 @@ describe('Browser', () => {
[C,test/client.js,387]                         });
[C,test/client.js,388]                     });
[C,test/client.js,389]                 });
[C,test/client.js,390]             });
[C,test/client.js,391] 
[C,test/client.js,392]             it('limits retries', (done) => {
[C,test/client.js,393] 
[C,test/client.js,394]                 const server = new Hapi.Server();
[C,test/client.js,395]                 server.connection();
[C,test/client.js,396]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,397] 
[C,test/client.js,398]                     expect(err).to.not.exist();
[C,test/client.js,399] 
[C,test/client.js,400]                     server.start((err) => {
[C,test/client.js,401] 
[A,test/client.js,402]                         expect(err).to.not.exist();
[C,test/client.js,403]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,404] 
[C,test/client.js,405]                         let c = 0;
[C,test/client.js,406]                         client.onConnect = function () {
[C,test/client.js,407] 
[C,test/client.js,408]                             ++c;
[C,test/client.js,409]                             client._ws.close();
[C,test/client.js,410]                         };
[C,test/client.js,411] 
[C,test/client.js,412]                         client.connect({ delay: 5, maxDelay: 10, retries: 2 }, () => {
[C,test/client.js,413] 
[C,test/client.js,414]                             setTimeout(() => {
[C,test/client.js,415] 
[C,test/client.js,416]                                 expect(c).to.equal(3);
[C,test/client.js,417]                                 server.stop(done);
@@ -404,30 +419,31 @@ describe('Browser', () => {
[C,test/client.js,419]                         });
[C,test/client.js,420]                     });
[C,test/client.js,421]                 });
[C,test/client.js,422]             });
[C,test/client.js,423] 
[C,test/client.js,424]             it('aborts reconnect if disconnect is called in between attempts', (done) => {
[C,test/client.js,425] 
[C,test/client.js,426]                 const server = new Hapi.Server();
[C,test/client.js,427]                 server.connection();
[C,test/client.js,428]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,429] 
[C,test/client.js,430]                     expect(err).to.not.exist();
[C,test/client.js,431] 
[C,test/client.js,432]                     server.start((err) => {
[C,test/client.js,433] 
[A,test/client.js,434]                         expect(err).to.not.exist();
[C,test/client.js,435]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,436] 
[C,test/client.js,437]                         let c = 0;
[C,test/client.js,438]                         client.onConnect = function () {
[C,test/client.js,439] 
[C,test/client.js,440]                             ++c;
[C,test/client.js,441]                             client._ws.close();
[C,test/client.js,442] 
[C,test/client.js,443]                             if (c === 1) {
[C,test/client.js,444]                                 setTimeout(() => {
[C,test/client.js,445] 
[C,test/client.js,446]                                     client.disconnect();
[C,test/client.js,447]                                 }, 5);
[C,test/client.js,448] 
[C,test/client.js,449]                                 setTimeout(() => {
@@ -453,30 +469,31 @@ describe('Browser', () => {
[C,test/client.js,469]                 server.register({ register: Nes, options: { auth: false, headers: '*' } }, (err) => {
[C,test/client.js,470] 
[C,test/client.js,471]                     expect(err).to.not.exist();
[C,test/client.js,472] 
[C,test/client.js,473]                     server.route({
[C,test/client.js,474]                         method: 'GET',
[C,test/client.js,475]                         path: '/',
[C,test/client.js,476]                         handler: function (request, reply) {
[C,test/client.js,477] 
[C,test/client.js,478]                             return reply('hello');
[C,test/client.js,479]                         }
[C,test/client.js,480]                     });
[C,test/client.js,481] 
[C,test/client.js,482]                     server.start((err) => {
[C,test/client.js,483] 
[A,test/client.js,484]                         expect(err).to.not.exist();
[C,test/client.js,485]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,486]                         client.connect(() => {
[C,test/client.js,487] 
[C,test/client.js,488]                             client.request({ path: '/' }, (err, payload, statusCode, headers) => {
[C,test/client.js,489] 
[C,test/client.js,490]                                 expect(err).to.not.exist();
[C,test/client.js,491]                                 expect(payload).to.equal('hello');
[C,test/client.js,492]                                 expect(statusCode).to.equal(200);
[C,test/client.js,493]                                 expect(headers).to.contain({ 'content-type': 'text/html; charset=utf-8' });
[C,test/client.js,494] 
[C,test/client.js,495]                                 client.disconnect();
[C,test/client.js,496]                                 server.stop(done);
[C,test/client.js,497]                             });
[C,test/client.js,498]                         });
[C,test/client.js,499]                     });
@@ -502,30 +519,31 @@ describe('Browser', () => {
[C,test/client.js,519]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,520] 
[C,test/client.js,521]                     expect(err).to.not.exist();
[C,test/client.js,522] 
[C,test/client.js,523]                     server.route({
[C,test/client.js,524]                         method: 'POST',
[C,test/client.js,525]                         path: '/',
[C,test/client.js,526]                         handler: function (request, reply) {
[C,test/client.js,527] 
[C,test/client.js,528]                             return reply('hello');
[C,test/client.js,529]                         }
[C,test/client.js,530]                     });
[C,test/client.js,531] 
[C,test/client.js,532]                     server.start((err) => {
[C,test/client.js,533] 
[A,test/client.js,534]                         expect(err).to.not.exist();
[C,test/client.js,535]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,536]                         client.connect(() => {
[C,test/client.js,537] 
[C,test/client.js,538]                             const a = { b: 1 };
[C,test/client.js,539]                             a.a = a;
[C,test/client.js,540] 
[C,test/client.js,541]                             client.request({ method: 'POST', path: '/', payload: a }, (err, payload, statusCode, headers) => {
[C,test/client.js,542] 
[C,test/client.js,543]                                 expect(err).to.exist();
[C,test/client.js,544]                                 expect(err.message).to.equal('Converting circular structure to JSON');
[C,test/client.js,545]                                 client.disconnect();
[C,test/client.js,546]                                 server.stop(done);
[C,test/client.js,547]                             });
[C,test/client.js,548]                         });
[C,test/client.js,549]                     });
@@ -539,30 +557,31 @@ describe('Browser', () => {
[C,test/client.js,557]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,558] 
[C,test/client.js,559]                     expect(err).to.not.exist();
[C,test/client.js,560] 
[C,test/client.js,561]                     server.route({
[C,test/client.js,562]                         method: 'POST',
[C,test/client.js,563]                         path: '/',
[C,test/client.js,564]                         handler: function (request, reply) {
[C,test/client.js,565] 
[C,test/client.js,566]                             return reply('hello');
[C,test/client.js,567]                         }
[C,test/client.js,568]                     });
[C,test/client.js,569] 
[C,test/client.js,570]                     server.start((err) => {
[C,test/client.js,571] 
[A,test/client.js,572]                         expect(err).to.not.exist();
[C,test/client.js,573]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,574]                         client.connect(() => {
[C,test/client.js,575] 
[C,test/client.js,576]                             client._ws.send = function () {
[C,test/client.js,577] 
[C,test/client.js,578]                                 throw new Error('boom');
[C,test/client.js,579]                             };
[C,test/client.js,580] 
[C,test/client.js,581]                             client.request({ method: 'POST', path: '/', payload: 'a' }, (err, payload, statusCode, headers) => {
[C,test/client.js,582] 
[C,test/client.js,583]                                 expect(err).to.exist();
[C,test/client.js,584]                                 expect(err.message).to.equal('boom');
[C,test/client.js,585]                                 client.disconnect();
[C,test/client.js,586]                                 server.stop(done);
[C,test/client.js,587]                             });
@@ -580,30 +599,31 @@ describe('Browser', () => {
[C,test/client.js,599] 
[C,test/client.js,600]                     setTimeout(() => {
[C,test/client.js,601] 
[C,test/client.js,602]                         return reply('hello');
[C,test/client.js,603]                     }, 20);
[C,test/client.js,604]                 };
[C,test/client.js,605] 
[C,test/client.js,606]                 const server = new Hapi.Server();
[C,test/client.js,607]                 server.connection();
[C,test/client.js,608]                 server.register({ register: Nes, options: { onMessage: onMessage } }, (err) => {
[C,test/client.js,609] 
[C,test/client.js,610]                     expect(err).to.not.exist();
[C,test/client.js,611] 
[C,test/client.js,612]                     server.start((err) => {
[C,test/client.js,613] 
[A,test/client.js,614]                         expect(err).to.not.exist();
[C,test/client.js,615]                         const client = new Nes.Client('http://localhost:' + server.info.port, { timeout: 10 });
[C,test/client.js,616]                         client.connect(() => {
[C,test/client.js,617] 
[C,test/client.js,618]                             client.message('winning', (err, response) => {
[C,test/client.js,619] 
[C,test/client.js,620]                                 expect(err).to.exist();
[C,test/client.js,621]                                 expect(err.message).to.equal('Request timed out');
[C,test/client.js,622]                                 expect(response).to.not.exist();
[C,test/client.js,623] 
[C,test/client.js,624]                                 setTimeout(() => {
[C,test/client.js,625] 
[C,test/client.js,626]                                     client.disconnect();
[C,test/client.js,627]                                     server.stop(done);
[C,test/client.js,628]                                 }, 20);
[C,test/client.js,629]                             });
@@ -613,30 +633,31 @@ describe('Browser', () => {
[C,test/client.js,633]             });
[C,test/client.js,634]         });
[C,test/client.js,635] 
[C,test/client.js,636]         describe('_send()', () => {
[C,test/client.js,637] 
[C,test/client.js,638]             it('catches send error without tracking', (done) => {
[C,test/client.js,639] 
[C,test/client.js,640]                 const server = new Hapi.Server();
[C,test/client.js,641]                 server.connection();
[C,test/client.js,642]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,643] 
[C,test/client.js,644]                     expect(err).to.not.exist();
[C,test/client.js,645] 
[C,test/client.js,646]                     server.start((err) => {
[C,test/client.js,647] 
[A,test/client.js,648]                         expect(err).to.not.exist();
[C,test/client.js,649]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,650]                         client.connect(() => {
[C,test/client.js,651] 
[C,test/client.js,652]                             client._ws.send = function () {
[C,test/client.js,653] 
[C,test/client.js,654]                                 throw new Error('failed');
[C,test/client.js,655]                             };
[C,test/client.js,656] 
[C,test/client.js,657]                             client._send({}, false, (err) => {
[C,test/client.js,658] 
[C,test/client.js,659]                                 expect(err).to.exist();
[C,test/client.js,660]                                 expect(err.message).to.equal('failed');
[C,test/client.js,661] 
[C,test/client.js,662]                                 client.disconnect();
[C,test/client.js,663]                                 server.stop(done);
@@ -664,30 +685,31 @@ describe('Browser', () => {
[C,test/client.js,685] 
[C,test/client.js,686]                             request.connection.plugins.nes._listener._sockets.forEach((socket) => {
[C,test/client.js,687] 
[C,test/client.js,688]                                 socket._ws.send('{');
[C,test/client.js,689]                             });
[C,test/client.js,690] 
[C,test/client.js,691]                             setTimeout(() => {
[C,test/client.js,692] 
[C,test/client.js,693]                                 return reply('hello');
[C,test/client.js,694]                             }, 10);
[C,test/client.js,695]                         }
[C,test/client.js,696]                     });
[C,test/client.js,697] 
[C,test/client.js,698]                     server.start((err) => {
[C,test/client.js,699] 
[A,test/client.js,700]                         expect(err).to.not.exist();
[C,test/client.js,701]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,702] 
[C,test/client.js,703]                         let logged;
[C,test/client.js,704]                         client.onError = function (err) {
[C,test/client.js,705] 
[C,test/client.js,706]                             logged = err.message;
[C,test/client.js,707]                         };
[C,test/client.js,708] 
[C,test/client.js,709]                         client.connect(() => {
[C,test/client.js,710] 
[C,test/client.js,711]                             client.request('/', (err, payload, statusCode, headers) => {
[C,test/client.js,712] 
[C,test/client.js,713]                                 expect(err).to.not.exist();
[C,test/client.js,714]                                 expect(logged).to.equal('Unexpected end of input');
[C,test/client.js,715] 
@@ -714,30 +736,31 @@ describe('Browser', () => {
[C,test/client.js,736] 
[C,test/client.js,737]                             request.connection.plugins.nes._listener._sockets.forEach((socket) => {
[C,test/client.js,738] 
[C,test/client.js,739]                                 socket._ws.send('{"id":100,"type":"response","statusCode":200,"payload":"hello","headers":{}}');
[C,test/client.js,740]                             });
[C,test/client.js,741] 
[C,test/client.js,742]                             setTimeout(() => {
[C,test/client.js,743] 
[C,test/client.js,744]                                 return reply('hello');
[C,test/client.js,745]                             }, 10);
[C,test/client.js,746]                         }
[C,test/client.js,747]                     });
[C,test/client.js,748] 
[C,test/client.js,749]                     server.start((err) => {
[C,test/client.js,750] 
[A,test/client.js,751]                         expect(err).to.not.exist();
[C,test/client.js,752]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,753] 
[C,test/client.js,754]                         let logged;
[C,test/client.js,755]                         client.onError = function (err) {
[C,test/client.js,756] 
[C,test/client.js,757]                             logged = err.message;
[C,test/client.js,758]                         };
[C,test/client.js,759] 
[C,test/client.js,760]                         client.connect(() => {
[C,test/client.js,761] 
[C,test/client.js,762]                             client.request('/', (err, payload, statusCode, headers) => {
[C,test/client.js,763] 
[C,test/client.js,764]                                 expect(err).to.not.exist();
[C,test/client.js,765]                                 expect(logged).to.equal('Received response for unknown request');
[C,test/client.js,766] 
@@ -764,70 +787,75 @@ describe('Browser', () => {
[C,test/client.js,787] 
[C,test/client.js,788]                             request.connection.plugins.nes._listener._sockets.forEach((socket) => {
[C,test/client.js,789] 
[C,test/client.js,790]                                 socket._ws.send('{"id":2,"type":"unknown","statusCode":200,"payload":"hello","headers":{}}');
[C,test/client.js,791]                             });
[C,test/client.js,792] 
[C,test/client.js,793]                             setTimeout(() => {
[C,test/client.js,794] 
[C,test/client.js,795]                                 return reply('hello');
[C,test/client.js,796]                             }, 10);
[C,test/client.js,797]                         }
[C,test/client.js,798]                     });
[C,test/client.js,799] 
[C,test/client.js,800]                     server.start((err) => {
[C,test/client.js,801] 
[A,test/client.js,802]                         expect(err).to.not.exist();
[C,test/client.js,803]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,804] 
[C,test/client.js,805]                         let logged;
[C,test/client.js,806]                         client.onError = function (err) {
[C,test/client.js,807] 
[C,test/client.js,808]                             if (!logged) {
[C,test/client.js,809]                                 logged = err.message;
[C,test/client.js,810]                                 return;
[C,test/client.js,811]                             }
[C,test/client.js,812] 
[C,test/client.js,813]                             expect(logged).to.equal('Received unknown response type: unknown');
[C,test/client.js,814]                             expect(err.message).to.equal('Received response for unknown request');
[C,test/client.js,815] 
[C,test/client.js,816]                             client.disconnect();
[C,test/client.js,817]                             server.stop(done);
[C,test/client.js,818]                         };
[C,test/client.js,819] 
[C,test/client.js,820]                         client.connect(() => {
[C,test/client.js,821] 
[D,test/client.js,798]                             client.request('/', (err, payload, statusCode, headers) => { });
[A,test/client.js,822]                             client.request('/', (err, payload, statusCode, headers) => {
[A,test/client.js,823] 
[A,test/client.js,824]                                 expect(err).to.not.exist();
[A,test/client.js,825]                             });
[C,test/client.js,826]                         });
[C,test/client.js,827]                     });
[C,test/client.js,828]                 });
[C,test/client.js,829]             });
[C,test/client.js,830]         });
[C,test/client.js,831] 
[C,test/client.js,832]         describe('subscribe()', () => {
[C,test/client.js,833] 
[C,test/client.js,834]             it('subscribes to a path', (done) => {
[C,test/client.js,835] 
[C,test/client.js,836]                 const server = new Hapi.Server();
[C,test/client.js,837]                 server.connection();
[C,test/client.js,838]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,839] 
[C,test/client.js,840]                     expect(err).to.not.exist();
[C,test/client.js,841] 
[C,test/client.js,842]                     server.subscription('/', {});
[C,test/client.js,843] 
[C,test/client.js,844]                     server.start((err) => {
[C,test/client.js,845] 
[A,test/client.js,846]                         expect(err).to.not.exist();
[C,test/client.js,847]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,848]                         client.connect(() => {
[C,test/client.js,849] 
[C,test/client.js,850]                             const handler = (update) => {
[C,test/client.js,851] 
[C,test/client.js,852]                                 expect(client.subscriptions()).to.deep.equal(['/']);
[C,test/client.js,853]                                 expect(update).to.equal('heya');
[C,test/client.js,854]                                 client.disconnect();
[C,test/client.js,855]                                 server.stop(done);
[C,test/client.js,856]                             };
[C,test/client.js,857] 
[C,test/client.js,858]                             client.subscribe('/', handler, (err) => {
[C,test/client.js,859] 
[C,test/client.js,860]                                 expect(err).to.not.exist();
[C,test/client.js,861]                                 server.publish('/', 'heya');
@@ -835,30 +863,31 @@ describe('Browser', () => {
[C,test/client.js,863]                         });
[C,test/client.js,864]                     });
[C,test/client.js,865]                 });
[C,test/client.js,866]             });
[C,test/client.js,867] 
[C,test/client.js,868]             it('subscribes to a unknown path (pre connect)', (done) => {
[C,test/client.js,869] 
[C,test/client.js,870]                 const server = new Hapi.Server();
[C,test/client.js,871]                 server.connection();
[C,test/client.js,872]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,873] 
[C,test/client.js,874]                     expect(err).to.not.exist();
[C,test/client.js,875] 
[C,test/client.js,876]                     server.start((err) => {
[C,test/client.js,877] 
[A,test/client.js,878]                         expect(err).to.not.exist();
[C,test/client.js,879]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,880] 
[C,test/client.js,881]                         client.subscribe('/b', Hoek.ignore, (err) => {
[C,test/client.js,882] 
[C,test/client.js,883]                             expect(err).to.not.exist();
[C,test/client.js,884] 
[C,test/client.js,885]                             client.connect((err) => {
[C,test/client.js,886] 
[C,test/client.js,887]                                 expect(err).to.exist();
[C,test/client.js,888]                                 expect(err.message).to.equal('Subscription not found');
[C,test/client.js,889]                                 expect(err.statusCode).to.equal(404);
[C,test/client.js,890]                                 expect(client.subscriptions()).to.be.empty();
[C,test/client.js,891] 
[C,test/client.js,892]                                 client.disconnect();
[C,test/client.js,893]                                 server.stop(done);
@@ -868,30 +897,31 @@ describe('Browser', () => {
[C,test/client.js,897]                 });
[C,test/client.js,898]             });
[C,test/client.js,899] 
[C,test/client.js,900]             it('subscribes to a path (pre connect)', (done) => {
[C,test/client.js,901] 
[C,test/client.js,902]                 const server = new Hapi.Server();
[C,test/client.js,903]                 server.connection();
[C,test/client.js,904]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,905] 
[C,test/client.js,906]                     expect(err).to.not.exist();
[C,test/client.js,907] 
[C,test/client.js,908]                     server.subscription('/');
[C,test/client.js,909] 
[C,test/client.js,910]                     server.start((err) => {
[C,test/client.js,911] 
[A,test/client.js,912]                         expect(err).to.not.exist();
[C,test/client.js,913]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,914] 
[C,test/client.js,915]                         const handler = (update) => {
[C,test/client.js,916] 
[C,test/client.js,917]                             expect(update).to.equal('heya');
[C,test/client.js,918]                             client.disconnect();
[C,test/client.js,919]                             server.stop(done);
[C,test/client.js,920]                         };
[C,test/client.js,921] 
[C,test/client.js,922]                         client.subscribe('/', handler, (err) => {
[C,test/client.js,923] 
[C,test/client.js,924]                             expect(err).to.not.exist();
[C,test/client.js,925] 
[C,test/client.js,926]                             client.connect((err) => {
[C,test/client.js,927] 
@@ -903,30 +933,31 @@ describe('Browser', () => {
[C,test/client.js,933]                 });
[C,test/client.js,934]             });
[C,test/client.js,935] 
[C,test/client.js,936]             it('manages multiple subscriptions', (done) => {
[C,test/client.js,937] 
[C,test/client.js,938]                 const server = new Hapi.Server();
[C,test/client.js,939]                 server.connection();
[C,test/client.js,940]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,941] 
[C,test/client.js,942]                     expect(err).to.not.exist();
[C,test/client.js,943] 
[C,test/client.js,944]                     server.subscription('/');
[C,test/client.js,945] 
[C,test/client.js,946]                     server.start((err) => {
[C,test/client.js,947] 
[A,test/client.js,948]                         expect(err).to.not.exist();
[C,test/client.js,949]                         const client1 = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,950]                         const client2 = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,951] 
[C,test/client.js,952]                         client1.connect((err) => {
[C,test/client.js,953] 
[C,test/client.js,954]                             expect(err).to.not.exist();
[C,test/client.js,955]                             client2.connect((err) => {
[C,test/client.js,956] 
[C,test/client.js,957]                                 expect(err).to.not.exist();
[C,test/client.js,958]                                 const handler = (update) => {
[C,test/client.js,959] 
[C,test/client.js,960]                                     expect(update).to.equal('heya');
[C,test/client.js,961]                                     client1.disconnect();
[C,test/client.js,962]                                     server.stop(done);
[C,test/client.js,963]                                 };
@@ -952,127 +983,134 @@ describe('Browser', () => {
[C,test/client.js,983]                 });
[C,test/client.js,984]             });
[C,test/client.js,985] 
[C,test/client.js,986]             it('ignores publish to a unknown path', (done) => {
[C,test/client.js,987] 
[C,test/client.js,988]                 const server = new Hapi.Server();
[C,test/client.js,989]                 server.connection();
[C,test/client.js,990]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,991] 
[C,test/client.js,992]                     expect(err).to.not.exist();
[C,test/client.js,993] 
[C,test/client.js,994]                     server.subscription('/');
[C,test/client.js,995] 
[C,test/client.js,996]                     server.start((err) => {
[C,test/client.js,997] 
[A,test/client.js,998]                         expect(err).to.not.exist();
[C,test/client.js,999]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1000]                         client.connect(() => {
[C,test/client.js,1001] 
[C,test/client.js,1002]                             client.subscribe('/', Hoek.ignore, (err) => {
[C,test/client.js,1003] 
[A,test/client.js,1004]                                 expect(err).to.not.exist();
[C,test/client.js,1005]                                 delete client._subscriptions['/'];
[C,test/client.js,1006] 
[C,test/client.js,1007]                                 server.publish('/', 'heya');
[C,test/client.js,1008]                                 setTimeout(() => {
[C,test/client.js,1009] 
[C,test/client.js,1010]                                     client.disconnect();
[C,test/client.js,1011]                                     server.stop(done);
[C,test/client.js,1012]                                 }, 10);
[C,test/client.js,1013]                             });
[C,test/client.js,1014]                         });
[C,test/client.js,1015]                     });
[C,test/client.js,1016]                 });
[C,test/client.js,1017]             });
[C,test/client.js,1018] 
[C,test/client.js,1019]             it('errors on unknown path', (done) => {
[C,test/client.js,1020] 
[C,test/client.js,1021]                 const server = new Hapi.Server();
[C,test/client.js,1022]                 server.connection();
[C,test/client.js,1023]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,1024] 
[C,test/client.js,1025]                     expect(err).to.not.exist();
[C,test/client.js,1026] 
[C,test/client.js,1027]                     server.start((err) => {
[C,test/client.js,1028] 
[A,test/client.js,1029]                         expect(err).to.not.exist();
[C,test/client.js,1030]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1031]                         client.connect(() => {
[C,test/client.js,1032] 
[C,test/client.js,1033]                             client.subscribe('/', Hoek.ignore, (err) => {
[C,test/client.js,1034] 
[C,test/client.js,1035]                                 expect(err).to.exist();
[C,test/client.js,1036]                                 expect(err.message).to.equal('Subscription not found');
[C,test/client.js,1037]                                 client.disconnect();
[C,test/client.js,1038]                                 server.stop(done);
[C,test/client.js,1039]                             });
[C,test/client.js,1040]                         });
[C,test/client.js,1041]                     });
[C,test/client.js,1042]                 });
[C,test/client.js,1043]             });
[C,test/client.js,1044] 
[C,test/client.js,1045]             it('subscribes and immediately unsubscribe to a path (all handlers)', (done) => {
[C,test/client.js,1046] 
[C,test/client.js,1047]                 const server = new Hapi.Server();
[C,test/client.js,1048]                 server.connection();
[C,test/client.js,1049]                 server.register({ register: Nes, options: { auth: false } }, (err) => {
[C,test/client.js,1050] 
[C,test/client.js,1051]                     expect(err).to.not.exist();
[C,test/client.js,1052] 
[C,test/client.js,1053]                     server.start((err) => {
[C,test/client.js,1054] 
[A,test/client.js,1055]                         expect(err).to.not.exist();
[C,test/client.js,1056]                         const client = new Nes.Client('http://localhost:' + server.info.port);
[C,test/client.js,1057]                         client.connect(() => {
[C,test/client.js,1058] 
[C,test/client.js,1059]                             const handler = (update) => {
[C,test/client.js,1060] 
[C,test/client.js,1061]                                 throw new Error('Must not be called');
[C,test/client.js,1062]                             };
[C,test/client.js,1063] 
[C,test/client.js,1064]                             client.subscribe('/', handler, (err) => {
[C,test/client.js,1065] 
[A,test/client.js,1066]                                 expect(err).to.exist();
[C,test/client.js,1067]                                 client.unsubscribe('/');
[C,test/client.js,1068] 
[C,test/client.js,1069]                                 setTimeout(() => {
[C,test/client.js,1070] 
[C,test/client.js,1071]                                     client.disconnect();
[C,test/client.js,1072]                                     server.stop(done);
[C,test/client.js,1073]                                 }, 20);
[C,test/client.js,1074]                             });
[C,test/client.js,1075]                         });
[C,test/client.js,1076]                     });
[C,test/client.js,1077]                 });
[C,test/client.js,1078]             });
[C,test/client.js,1079] 
[C,test/client.js,1080]             it('subscribes and immediately unsubscribe to a path (single handler)', (done) => {
[C,test/client.js,1081] 
[C,test/client.js,1082]                 const server = new Hapi.Server();
